apps/air-discount-scheme/backend/src/app/modules/flight/flight.controller.ts

```diff
@@ -94,15 +164,74 @@ export class PublicFlightController {
     if (flightLegsLeft < flight.flightLegs.length) {
       throw new ForbiddenException('Flight leg quota is exceeded')
     }
+
+    let connectingFlight = false
+    let connectingId = undefined
+
+    const flightLegCount = flight.flightLegs.length
+
+    const incomingLeg = {
+      origin: flight.flightLegs[0].origin,
+      destination: flight.flightLegs[flightLegCount - 1].destination,
```

Comment:
It is probably better to take the first flightleg here also, so we can check in the flightservice if the destination matches the ALLOWED_CONNECTING_FLIGHT_CODES

Reply:
It is assumed that flightLegs for a single flight are given chronologically, right? 
That's why the destination is pulled from `flightLegs[flightLegs.length - 1].destination`
Does the intermediate airport matter in the cases where a flight has more than one flightLeg?

Reply:
Re-orders flightLegs chronologically. 
Checks whether firstLeg origin is the same as lastLeg destination to detect round-trips.

Functionality remains the same if no round-trip detected.

If roundtrip detected:
The flightLeg check was going through every leg of the connectedId (Reykjavík flight)  so we can send the lastLeg to that too when roundtrips are detected.
That way we check that when your returnTrip flight from norlandair returns to Akureyri, whether it has a corresponding return flightLeg from icelandair.